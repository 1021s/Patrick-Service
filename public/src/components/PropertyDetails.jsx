/* eslint-disable react/prop-types */
import React from 'react';
import ListHelper from './ListHelper';

const PropertyDetails = ({ listing }) => (
  <div>
    <div>
      <div className="type">Parking</div>
      <ul>
        {typeof listing === 'object' && Object.keys(listing.parking).map((key) => (
          <ListHelper key={key} entry={key} value={listing.parking[key]} />
        ))}
      </ul>
    </div>
    <div>
      <div className="type">Property</div>
      <ul>
        {typeof listing === 'object' && Object.keys(listing.property).map((key) => (
          <ListHelper key={key} entry={key} value={listing.property[key]} />
        ))}
      </ul>
    </div>
    <div>
      <div className="type">Lot</div>
      <ul>
        {typeof listing === 'object' && Object.keys(listing.lot).map((key) => (
          <ListHelper key={key} entry={key} value={listing.lot[key]} />
        ))}
      </ul>
    </div>
    <div>
      <div className="type">Other property information</div>
      <ul>
        {typeof listing === 'object' && Object.keys(listing.otherPropertyInformation).map((key) => (
          <ListHelper key={key} entry={key} value={listing.otherPropertyInformation[key]} />
        ))}
      </ul>
    </div>
  </div>
);

export default PropertyDetails;
