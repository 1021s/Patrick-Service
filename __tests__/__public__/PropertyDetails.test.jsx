/* eslint-env jest */
/* eslint-env node */
import 'jsdom-global/register';
import React from 'react';
import '../../setupTests';
import { mount, shallow } from 'enzyme';
import PropertyDetails from '../../public/src/components/PropertyDetails';

describe('MyComponent', () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<PropertyDetails debug />);
    expect(component).toMatchSnapshot();
  });

  it('should render correctly with no props', () => {
    const component = mount(<PropertyDetails />);
    expect(component).toMatchSnapshot();
  });
});
