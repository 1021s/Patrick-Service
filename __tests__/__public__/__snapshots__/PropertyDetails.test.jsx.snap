// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyComponent should render correctly in "debug" mode 1`] = `ShallowWrapper {}`;

exports[`MyComponent should render correctly with no props 1`] = `ReactWrapper {}`;
